File: readability/readability.c
#include "readability.h"

int main(void)
{
    // gets input from user
    string in = get_string("Text: ");

    // gets values from functions
    int letters = get_letters(in);
    int words = get_words(in);
    int sentences = get_sentences(in);

    // printf("%i\n", letters);
    // printf("%i\n", words);
    // printf("%i\n", sentences);

    // prints out grade level
    float L = ((float)letters / words)*100;
    printf("%f\n", L);
    float S = ((float)sentences / words)*100;
    printf("%f\n", S);
    float index = (0.0588 * L) - (0.296 * S) - 15.8;
    printf("%f\n", index);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    } else if (index >= 16)
    {
        printf("Grade 16+\n");
    } else { printf("Grade %i\n", (int)index); }
}

// finds the number of letters
int get_letters(string in)
{
    int letters = 0;
    int i = 0;

    while (in[i] != '\0')
    {
        if (isalpha(in[i]))
        {
            letters++;
        }
        i++;
    }

    return letters;
}

// finds the number of words
int get_words(string in)
{
    int words = 0;
    int i = 0;

    while (in[i] != '\0')
    {
        if (in[i] == ' ')
        {
            words++;
        }
        i++;
    }

    words++;
    return words;
}

// finds the number of sentences
int get_sentences(string in)
{
    int sentences = 0;
    int i = 0;

    while (in[i] != '\0')
    {
        bool isSentence = (in[i] == '.' || in[i] == '!' || in[i] == '?');
        if (isSentence)
        {
            sentences++;
        }
        i++;
    }

    return sentences;
}

File: sentimental-readability/readability.py
from cs50 import *

# finds the number of letters
def get_letters(text_in):
    letters = 0
    i = 0

    while text_in[i] != '\0':
        if text_in[i].isalpha():
            letters += 1
        i += 1

    return letters

# finds the number of words
def get_words(text_in):
    words = 0
    i = 0

    while text_in[i] != '\0':
        if text_in[i] == ' ':
            words += 1
        i += 1

    words += 1
    return words

# finds the number of sentences
def get_sentences(text_in):
    sentences = 0
    c = 0

    while text_in[c] != '\0':
        isSentence = text_in[c] == '.' or text_in[c] == '!' or text_in[c] == '?'
        if isSentence:
            sentences += 1
        c += 1

    return sentences

def main():
    # gets input from user
    text_in = get_string("Text: ")

    # gets values from functions
    letters = get_letters(text_in)
    words = get_words(text_in)
    sentences = get_sentences(text_in)

    # prints out grade level
    L = (float(letters / words)) * 100
    S = (float(sentences / words)) * 100
    index = (0.0588 * L) - (0.296 * S) - 15.8

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {int(round(index))}")

main()